### PR TITLE
Don't cancel pending visualization's upserts

### LINKED ISSUE #8801: After loading project, some visualization does not load.
When opening the following project, the lowest visualizations seem to be stuck at loading state:

[Bookclub_17_01.zip](https://github.com/enso-org/enso/files/13975479/Bookclub_17_01.zip)

We receive some "Visualization not found" log

Image

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

Uniqueness check of `UpsertVisualizationJob` only involved expressionId. Apparently now GUI sends mutliple visualizations for the same expressions and expects all of them to exist. Since previously we would cancel duplicate jobs, this was problematic.

This change makes sure that uniqueness also takes into account visualization id. Fixed a few logs that were not passing arguments properly.

Closes #8801 

### Important Notes

I have not noticed any more problems with loading visualizations so the issue appears to be resolved with this change.
Added a unit test case that would previously fail due to cancellation of a job that upserts visualization.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ class UpsertVisualizationJob(
   override def equalsTo(that: UniqueJob[_]): Boolean =
     that match {
       case that: UpsertVisualizationJob =>
-        this.expressionId == that.expressionId
+        this.expressionId == that.expressionId && this.visualizationId == that.visualizationId
```
Comment (hubertp) [inline engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:56]: This was the root of the problem.

File: engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala [hunk 2]
```diff
@@ -2058,512 +1169,555 @@ class RuntimeVisualizationsTest
...
-      None,
-      Vector()
-    )
-    context.send(
-      Api.Request(requestId, Api.PushContextRequest(contextId, item1))
+    val attachVisualizationResponses = context.receiveN(2)
+
+    attachVisualizationResponses.filter(
+      _.payload.isInstanceOf[Api.VisualizationAttached]
+    ) shouldEqual List(
```
Comment (hubertp) [inline engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:1360]: This test would previously fail.

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:56]: Might deserve a simple unit test comparing two job instances for equality. The `RuntimeVisualizationsTest` may contain such check, but it is very hidden in the whole logic of the test.

**Review by JaroslavTulach — APPROVED** [review]
Mostly process oriented approval. I didn't try to understand the changes in `RuntimeVisualizationsTest`.

File: engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala (hunk 1 above)
Comment (hubertp) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:56]: Added

Comment (JaroslavTulach) [reply engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:56]: I see 1d8cd25e9315e3348d8700778105001903eb3b8d, thanks.

**Comment by hubertp** [comment]
Based on the last failures I basically think that `runtime/test` does not attempt to recompile tests in `runtime-instrument-common` on which it indirectly depends on. Then no wonder it does **not** have the required delays to simulate the problem.

**Pull request merged**